src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -72,26 +72,66 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
+            ResourceOperation operationType = ResourceOperation.NoOperation;
+
             if (existingObservation == null)
             {
                 var newObservation = GenerateObservation(config, observationGroup, identifier, ids);
                 result = await _client.CreateAsync(newObservation).ConfigureAwait(false);
-                _logger.LogMetric(IomtMetrics.FhirResourceSaved(ResourceType.Observation, ResourceOperation.Created), 1);
+                operationType = ResourceOperation.Created;
             }
             else
             {
+                Model.Observation mergedObservation = null;
+
                 var policyResult = await Policy<Model.Observation>
-                    .Handle<FhirOperationException>(ex => ex.Status == System.Net.HttpStatusCode.Conflict || ex.Status == System.Net.HttpStatusCode.PreconditionFailed)
-                    .RetryAsync(2, async (polyRes, attempt) =>
-                    {
-                        existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
-                    })
-                    .ExecuteAndCaptureAsync(async () =>
-                    {
-                        var mergedObservation = MergeObservation(config, existingObservation, observationGroup);
-                        return await _client.UpdateAsync(mergedObservation, versionAware: true).ConfigureAwait(false);
-                    }).ConfigureAwait(false);
+                     .Handle<FhirOperationException>(ex => ex.Status == System.Net.HttpStatusCode.Conflict || ex.Status == System.Net.HttpStatusCode.PreconditionFailed)
+                     .RetryAsync(2, async (polyRes, attempt) =>
+                     {
+                         // 409 Conflict or 412 Precondition Failed can occur if the Observation.meta.versionId does not match the update request.
+                         // This can happen if 2 independent processes are updating the same Observation simultaneously.
+                         // or
+                         // The update operation failed because the Observation no longer exists.
+                         // This can happen if a cached Observation was deleted from the FHIR Server.
+
+                         // Attempt to get the most recent version of the Observation.
+                         existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
+
+                         // If the Observation no longer exists on the FHIR Server, it was most likely deleted.
+                         if (existingObservation == null)
+                         {
+                             // Remove the Observation from the cache (this version no longer exists on the FHIR Server.
+                             _observationCache.Remove(cacheKey);
+
+                             // Create a new Observation.
+                             result = await _client.CreateAsync(mergedObservation).ConfigureAwait(false);

```

Comment:
Is `mergedObservation` `null` at this point? If so, won't that result in a blank Observation? Since you know that there is no Observation inside of the Fhir Server could we simply create a new Observation using the data passed into this method?

```c#
// Create a new Observation.
var newObservation = GenerateObservation(config, observationGroup, identifier, ids);
result = await _client.CreateAsync(newObservation).ConfigureAwait(false);
```

Reply:
mergedObservation here is not null because the ExecuteAndCaptureAsync block has already run one time, which performed the merge operation. I see your point though, where if an external actor deleted the Observation, we should not create a new Observation with the data that was previously deleted. I will update this to just create a new Observation.

Reply:
Agreed.  This should work pretty well if you follow the other suggestion to move the core update logic into the ExecuteAndCapture  block.